### PR TITLE
added spotless

### LINKED ISSUE #11: Setup Java Formatter Checks Using Checkstyle/Spotless in CI
Adding Checkstyle/Spotless validations to the CI will ensure that all code follows the formatting rules before being merged.

Checkstyle GitHub Action on marketplace - [actions/checkstyle-for-java](https://github.com/marketplace/actions/checkstyle-for-java)

Update in the README file the steps to run the formatter locally before pushing the code.

### PR DESCRIPTION
**Pull request by Taz03**
closes #11

### CODE REVIEW

**Comment by navin772** [comment]
Thanks for the PR @Taz03 , can you add some steps (even if it is single command) for formatting in the README file to run `spotless`.
Also, create a new CI that would check for formatting and pass/fail based on push/PR.

**Comment by Taz03** [comment]
ok it, mb i forgot to make this pr a draft

File: .github/workflows/format-check.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'oracle'
+
+      - name: Build with Gradle
+        run: ./gradlew spotlessCheck
```
Comment (Delta456) [inline .github/workflows/format-check.yml:24]: Add newline here

File: build.gradle [hunk 2]
```diff
@@ -17,18 +18,25 @@ dependencies {
...
 }
+
+spotless {
+    java {
+        importOrder()
+        removeUnusedImports()
+        googleJavaFormat().aosp()
+        formatAnnotations()
+    }
+}
```
Comment (Delta456) [inline build.gradle:42]: Add newline here too

File: .github/workflows/format-check.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Check Format
+
+on:
+  push:
+    branches: [ 'main' ]
```
Comment (harsha509) [inline .github/workflows/format-check.yml:5]: HI @Taz03 ,
Do we require ticks for branch name in here ?
will this work ?

```suggestion
    branches: [main]
```

File: .github/workflows/format-check.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
```
Comment (harsha509) [inline .github/workflows/format-check.yml:20]: just a minor suggestion,
Can you have distribution forst and java-version next ?

This is only for readability!

File: src/main/java/io/github/lambdatest/gradle/AppUploader.java [hunk 5]
```diff
@@ -19,35 +19,46 @@ public AppUploader(String username, String accessKey, String appFilePath) {
...
+                        Request request =
+                                new Request.Builder()
+                                        .url(Constants.API_URL)
+                                        .method("POST", body)
+                                        .addHeader(
+                                                "Authorization",
+                                                Credentials.basic(username, accessKey))
+                                        .build();
+                        try (Response response = client.newCall(request).execute()) {
+                            if (!response.isSuccessful())
```
Comment (harsha509) [inline src/main/java/io/github/lambdatest/gradle/AppUploader.java:46]: Just checking,
can we add a check if `response is successful and responseBody is null` 
@Delta456 , any thoughts on this ?

Comment (harsha509) [reply src/main/java/io/github/lambdatest/gradle/AppUploader.java:46]: Ah my bad, this is just the format changes!
Ignore above!

File: .github/workflows/format-check.yml (hunk 3 above)
Comment (Taz03) [reply .github/workflows/format-check.yml:5]: Yes those are required afaik, or I can expand the list and have `- main`, if u prefer that

File: .github/workflows/format-check.yml (hunk 4 above)
Comment (Taz03) [reply .github/workflows/format-check.yml:20]: Should be possible I'll do the change

File: .github/workflows/format-check.yml (hunk 3 above)
Comment (Delta456) [reply .github/workflows/format-check.yml:5]: I believe both will work!

image

Comment (Taz03) [reply .github/workflows/format-check.yml:5]: so which one do we want?

**Comment by navin772** [comment]
@Taz03 can you please resolve the conflicts?

File: .github/workflows/format-check.yml (hunk 3 above)
Comment (navin772) [reply .github/workflows/format-check.yml]: We can go with ` branches: [main]` as @harsha509 suggested. The ticks are not required.

File: .settings/org.eclipse.buildship.core.prefs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+arguments=--init-script /home/taz/.cache/jdtls/config/org.eclipse.osgi/148/0/.cp/gradle/init/init.gradle
```
Comment (harsha509) [inline .settings/org.eclipse.buildship.core.prefs:1]: are these required ?

Comment (Taz03) [reply .settings/org.eclipse.buildship.core.prefs]: no, im working on it

**Review by harsha509 — APPROVED** [review]
LGTM!

**Review by Delta456 — APPROVED** [review]
Good work @Taz03! Thanks for your contributions :) 

I will be merging this now 🌟

**Pull request merged**